packages/user_packages/iotcc_mqtt_dev.py

```diff
@@ -0,0 +1,186 @@
+    """
+    This UDM randomly return values in between 0 to 999 in order to simulate device metric
+    Use case specific device metric collection logic should be provided by user
+    :return:
+    """
+    return randint(0, 999)
+
+
+class PackageClass(LiotaPackage):
+    def run(self, registry):
+        """
+        The execution function of a liota package.
+        Acquires "iotcc_mqtt" and "iotcc_mqtt_edge_system" from registry then register five devices
+        and publishes device metrics to the DCC
+        :param registry: the instance of ResourceRegistryPerPackage of the package
+        :return:
+        """
+        from liota.entities.devices.simulated_device import SimulatedDevice
+        from liota.entities.metrics.metric import Metric
+        import copy
+
+        # Acquire resources from registry
+        self.iotcc = registry.get("iotcc_mqtt")
+        # Creating a copy of edge_system object to keep original object "clean"
+        self.iotcc_edge_system = copy.copy(registry.get("iotcc_mqtt_edge_system"))
+
+        self.reg_devices = []
+        num_devices = 5
+
+        for i in range(0, num_devices):
+            try:
+                # Register device
+                device = SimulatedDevice(socket.gethostname() + "-ChildDev" + str(i))
+                log.info("Registration Started for Device {0}".format(device.name))
+                # Device Registration attempts
+                reg_attempts = 0
+                # Started Device Registration attempts
+                while reg_attempts <= retry_attempts:
+                    try:
+                        reg_dev = self.iotcc.register(device)
+                        break
+                    except Exception, e:
+                        if reg_attempts == retry_attempts:
+                            raise
+                        reg_attempts += 1
+                        log.error(
+                            '{0} Device Registration Failed with following error - {1}'.format(device.name, str(e)))
+                        log.info('{0} Device Registration: Attempt: {1}'.format(device.name, str(reg_attempts)))
+                        time.sleep(delay_retries)
+
+                self.reg_devices.append(reg_dev)
+                # Attempts to set device relationship with edge system
+                relationship_attempts = 0
+                while relationship_attempts <= retry_attempts:
+                    try:
+                        self.iotcc.create_relationship(self.iotcc_edge_system, reg_dev)
+                        break
+                    except Exception, e:
+                        if relationship_attempts == retry_attempts:
+                            raise
+                        relationship_attempts += 1
+                        log.error(
+                            '{0} Device relationship with Edge System failed with following error - {1}'.format(
+                                device.name, str(e)))
+                        log.info(
```

Comment:
Doesn't have the Trying log. Other logs are usually starting with Trying word.

Reply:
Added